src/kernels/raw/finder/localKnownPathKernelSpecFinder.node.ts

```diff
@@ -21,14 +17,27 @@ import { IFileSystemNode } from '../../../platform/common/platform/types.node';
 import { IMemento, GLOBAL_MEMENTO } from '../../../platform/common/types';
 import { capturePerfTelemetry, Telemetry } from '../../../telemetry';
 import { sendKernelSpecTelemetry } from './helper';
+import { noop } from '../../../platform/common/utils/misc';
 
 /**
  * This class searches for kernels on the file system in well known paths documented by Jupyter.
  * This will return Python, Julia, R etc kernels.
  * Returns all kernels regardless of whether Python extension is installed or not.
  */
 @injectable()
-export class LocalKnownPathKernelSpecFinder extends LocalKernelSpecFinderBase {
+export class LocalKnownPathKernelSpecFinder extends LocalKernelSpecFinderBase<LocalKernelSpecConnectionMetadata> {
+    private _cachedKernels: LocalKernelSpecConnectionMetadata[] = [];
+    private _initialized?: Promise<void>;
+    public get initialized() {
+        return this.initialize();
+    }
+    private readonly _onDidChangeKernels = new EventEmitter<void>();
+    /**
+     * TODO: We can monitor the known kernel spec folders and files for changes and trigger the change event.
+     * Lets discuss with VS Code core if there are known perf issues.
+     * If there are, then there's no need to monitor these folders/files for now.
```

Comment:
This work will be done in a subsequent PR
as mentioend in standup, i want to get the framework ready (else getting everything in one PR would result in a lot of changes)